src/rule_definition_tools.jl

```diff
@@ -436,9 +444,7 @@ function _nondiff_rrule_expr(__source__, primal_sig_parts, primal_invoke)
     tup_expr = tuple_expression(primal_sig_parts)
     primal_name = first(primal_invoke.args)
     pullback_expr = @strip_linenos quote
-        function $(esc(propagator_name(primal_name, :pullback)))(@nospecialize(_))
-            return $(tup_expr)
-        end
+        NonDiffPullback($(tup_expr))
```

Comment:
I guess even simpler would be
```suggestion
        Returns($(tup_expr))
```

Reply:
That will change the method signature, though. Currently the pullback accepts a single argument, while `Returns` accepts any amount of arguments. Is that OK to change?